channeldb/kvdb/etcd/commit_queue_test.go

```diff
@@ -68,48 +69,51 @@ func TestCommitQueue(t *testing.T) {
 	defer cancel()
 
 	wg.Add(numCommits)
-	t1 := time.Now()
+
+	var block sync.Mutex
 
 	// Tx1: reads: key1, key2, writes: key3, conflict: none
-	q.Add(
-		commit("free", true),
+	// Since we simulate that the txn takes a long time, we'll add in a
+	// new goroutine and wait for the txn commit to start execution.
+	commitTx1 := make(chan struct{})
+	// Block the commit by holding the mutex.
+	block.Lock()
+	go q.Add(
+		commit("free", commitTx1, &block),
 		makeReadSet([]string{"key1", "key2"}),
 		makeWriteSet([]string{"key3"}),
 	)
-	// Tx2: reads: key1, key2, writes: key3, conflict: Tx1
-	q.Add(
-		commit("blocked1", false),
-		makeReadSet([]string{"key1", "key2"}),
-		makeWriteSet([]string{"key3"}),
-	)
-	// Tx3: reads: key1, writes: key4, conflict: none
+	<-commitTx1
+
+	// Tx2: reads: key1, key5, writes: key3, conflict: Tx1 (key3)
+	// We don't expect queue add to block as this txn will queue up after
+	// tx1.
 	q.Add(
-		commit("free", true),
```

Comment:
This test seems a bit weaker now, in that it doesn't actually test for transaction reordering. If the commit queue simply executed the transactions in the order they are added, the test would pass. The new version doesn't test that concurrent non-conflicting transactions are executed immediately.

I think it makes sense to keep the current transactions, possibly even making the test stronger by naming the transactions `free1` and `free2` to assert their order.